framework/source/class/qx/data/marshal/Json.js

```diff
@@ -83,7 +83,10 @@ qx.Class.define("qx.data.marshal.Json",
      */
     __jsonToHash : function (data, includeBubbleEvents)
     {
-      return Object.keys(data).sort().join('"')
+      if (this.__delegate && this.__delegate.getJsonHash) {
+        return this.__delegate.getJsonHash(data, includeBubbleEvents);
+      }
+      return Object.keys(data).sort().join('|')
```

Comment:
Since the hash function gives complete freedom I would not change the current default as it may have side effects in applications where people have been looking at the generated strings in their application code ...

Reply:
But the existing code leads to errors as shown in the example. Might not be relevant for existing user code, because they would have recognized the error, but for new ones. I consider this as a bug, which needs to be fixed. I wouldn't stick to backwards compability here, because it is a very rare use case. And users can use the delegate to go back to the old behaviour, if they really need to.

Reply:
true ... ok